File: src/define/message.go
package define

const (
	MSG_TYPE_ONLINE     = "online"
	MSG_TYPE_ACK_ONLINE = "ackonline"
	MSG_TYPE_OFFLINE    = "offline"
	MSG_TYPE_JOIN       = "join"
	MSG_TYPE_TALK       = "talk"

	MSG_TYPE_USER_ADD = "addUser"
)

type Message struct {
	MsgType  string
	From     string
	HeadImg  string
	To       string
	IsPublic bool
	Content  string
}

type AddUserMessage struct {
	UserName string
	Password string
}

type ToClientMessage struct {
	MsgType string
	Content string
}

File: www/js/message.js
/* global define */
define(function(require, exports, module){
    function Message() {
        this.MsgType = this.MSG_TYPE_TALK;
        this.To = 'all';
        this.IsToOne = false;
        this.Content = '';
    }
    
    Message.prototype.MSG_TYPE_TALK = 'talk';
    
    return Message;
});